scripts/mysql_system_tables_fix.sql

```diff
@@ -803,9 +803,9 @@ COMMIT;
 
 -- Add the privilege AUTHENTICATION_POLICY_ADMIN for every user who has the SYSTEM_VARIABLES_ADMIN privilege
 -- provided that there isn't a user who already has the privilege AUTHENTICATION_POLICY_ADMIN.
-SET @hadAuthenticationPolicyAdminPriv = (SELECT COUNT(*) FROM global_grants WHERE priv = 'AUTHENTICATION_POLICY_ADMIN');
+SET @hadAuthenticationPolicyAdminPriv = (SELECT COUNT(*) FROM global_grants WHERE priv = 'AUTHENTICATION_POLICY_ADMIN' AND user NOT IN ('mysql.session'));
```

Comment:
Why this change?

Reply:
@kamil-holubicki This is the change I mentioned when I did the preliminary merge as well.
The upstream bug which introduced the new change _Bug #33165861: WL#14183 new privileges not taken into account for upgrades_ and the comment says that if there isn't a user who already has AUTHENTICATION_POLICY_ADMIN privilege, the new privilege upgrade happens. But looking at the script, it seems like mysql.session user is NOT considered whenever privilege upgrade like these are made. So, I think mysql.session user having AUTHENTICATION_POLICY_ADMIN privilege should also NOT be considered here. What do you think? Should we report an upstream bug for it?